pkg/asset/handler/upload_form.go

```diff
@@ -61,23 +62,22 @@ func (h *UploadFormHandler) Handle(w http.ResponseWriter, r *http.Request) (err
 	// Verify signature
 	err = h.PresignProvider.Verify(r)
 	if err != nil {
-		err = skyerr.NewError(skyerr.BadRequest, err.Error())
 		return
 	}
 
 	contentType := r.Header.Get("Content-Type")
 	mediaType, params, err := mime.ParseMediaType(contentType)
 	if err != nil {
-		err = skyerr.NewError(skyerr.BadRequest, "invalid content-type")
+		err = skyerr.Invalid.WithReason("InvalidContentType").New("invalid content-type")
```

Comment:
These errors should be `BadRequest` instead of `Invalid`, since the server does not understand the request. I should clarify that `BadRequest` and `Invalid` is like syntactical and semantic error.

Also, there is not need to provide this level of details to client. I think it is better to use a generic reason `InvalidUploadBody` for these errors, since these errors are programming errors that would not be inspected at runtime.

Reply:
> I should clarify that BadRequest and Invalid is like syntactical and semantic error.

Thanks. Better to put this guideline in the doc.